libcontainer/cgroups/systemd/common.go

```diff
@@ -365,22 +365,24 @@ func systemdVersion(conn *systemdDbus.Conn) (int, error) {
 			return
 		}
 
-		// verStr is like "v245.4-1.fc32" (including quotes)
-		if !strings.HasPrefix(verStr, `"v`) {
-			versionErr = fmt.Errorf("can't parse version %s", verStr)
-			return
-		}
-		// remove `"v` prefix and everything after the first dot
-		ver, err := strconv.Atoi(strings.SplitN(verStr[2:], ".", 2)[0])
-		if err != nil {
-			versionErr = err
-		}
-		version = ver
+		version, versionErr = systemdVersionAtoi(verStr)
+		return
 	})
 
 	return version, versionErr
 }
 
+func systemdVersionAtoi(verStr string) (int, error) {
+	// verStr is either like "v245.4-1.fc32" or "245" (both including quotes)
+	// thus, we can unconditionally remove the leading "v, as well as
+	// everything after the `.`
+	ver, err := strconv.Atoi(strings.SplitN(strings.Trim(verStr, "\"v"), ".", 2)[0])
+	if err != nil {
+		err = errors.Wrapf(err, "can't parse version %s", verStr)
+	}
+	return ver, err
```

Comment:
you can actually just remove the `if` and use `return ver, errors.Wrapf(err, "can't parse version %s", verStr)` since `errors. Wrapf` returns nil if err is nil